app/validators/qc_receptions_factory_validator.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+# Failed validations return unprocessable_entity
+# These are being validated before any entity is created
+class QcReceptionsFactoryValidator < ActiveModel::Validator
+  def validate(record)
+    validations = %i[validate_qc_results_list validate_assay_type]
+
+    validations.each do |validation|
+      next if record.errors.present?
+
+      send(validation, record)
+    end
+  end
+
+  def validate_qc_results_list(record)
+    # Check if array is not empty
+    if record.qc_results_list.empty?
+      record.errors.add :qc_results_list, :empty_array
+      return
+    end
+    # Return if any object inside the qc_results_list array has values
+    return nil if record.qc_results_list.any? { |value| !value.empty? }
+
+    # Add error if the objects inside the qc_results_list array is empty
+    record.errors.add :qc_results_list, :empty_array
+    nil
```

Comment:
Think this section could be refactored a little. e.g is the list is empty or any value is empty, error. Otherwise return nil?